src/QMCDrivers/DMC/DMCBatched.h

```diff
@@ -47,19 +47,24 @@ class DMCBatched : public QMCDriverNew
   struct StateForThread
   {
     const QMCDriverInput& qmcdrv_input;
-    const DMCDriverInput& dmcdrv_input;
     const DriftModifierBase& drift_modifier;
     const MCPopulation& population;
     SFNBranch& branch_engine;
     IndexType recalculate_properties_period;
-    IndexType step            = -1;
+    const size_t steps_per_block;
```

Comment:
since `IndexType` is signed and `steps_per_block` is unsigned and we're likely to compare `step` and `steps_per_block` and this potentially an issue.  As is doing arithmatic operations with mixed signed and unsiged types.  I think steps_per_block should be signed.

Reply:
"steps_per_block" should be unsigned by nature.
```
for (int step = 0; step < steps_per_block_; ++step)
```
I have not heard of unsigned loop upper bound being an issue.

Reply:
The issue is the signed/unsigned comparison, which is not a great habit. We probably have several of them in the code which makes me wonder if compiler warnings have been disabled for this case -- usually there will be a warning.

Reply:
since we guarantee the counter `step` positive, there should be no concern. If a real case that matters, I would prefer changing to size_t or int depending on the actual need.

Reply:
Something being positive definite is not a good reason to use an unsigned type.  If you are going to do any arithmetic with it save some grief and use a signed type.